rest_api/sawtooth_rest_api/messaging.py

```diff
@@ -161,10 +180,21 @@ def __init__(self, socket, msg_router):
             content=message_content,
             message_type=message_type)
 
-        try:
-            await self._socket.send_multipart([message.SerializeToString()])
-        except asyncio.CancelledError:
-            raise
+        # Send the message. Backoff and retry in case of an error
+        backoff = _Backoff()
+
+        while True:
+            try:
+                await self._socket.send_multipart(
+                    [message.SerializeToString()])
+                break
+            except asyncio.CancelledError:
+                LOGGER.warning('ZMQ connection dropped. Backing off.')
```

Comment:
This should at minimum contain the backoff attempt and waiting period in the logging message.

Is it factual that "asyncio.CancelledError" is always equivalent to a ZMQ connection being dropped? If not (which seems possible), this would be a misleading logging statement in those other circumstances.

Reply:
I'll remove the log messages from here. That should address all these concerns.

Reply:
It indicates that the loop has been shutdown, not disconnect.